packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -775,20 +775,19 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
 
         if (checkTombstone && this.tombstoned) {
             const messageString = `Context is tombstoned! Call site [${callSite}]`;
-            const error = new DataCorruptionError(messageString, {
-                errorMessage: messageString,
-                ...safeTelemetryProps,
-            });
-
-            // Always log an error when tombstoned data store is used. However, throw an error only if
-            // throwOnTombstoneUsage is set.
-            const event = {
-                eventName: "GC_Tombstone_DataStore_Changed",
-                callSite,
-            };
-            sendGCTombstoneEvent(this.mc, event, this.clientDetails.type === summarizerClientType, this.pkg, error);
-            // Always log an error when tombstoned data store is used. However, throw an error only if
-            // throwOnTombstoneUsage is set and the client is not a summarizer.
+            const error = new DataCorruptionError(messageString, safeTelemetryProps);
+
+            sendGCTombstoneEvent(
+                this.mc.logger,
+                {
+                    eventName: "GC_Tombstone_DataStore_Changed",
+                    callSite,
+                },
+                this.throwOnTombstoneUsage /* logAsError */,
+                this.clientDetails.type === summarizerClientType,
+                this.pkg,
+                error,
+            );
             if (this.throwOnTombstoneUsage) {
                 throw error;
```

Comment:
We should update this to not throw right? Otherwise, even if data stores are revived by using the recovery scenario, using them will result in this error being thrown. Do you plan to do that in a separate PR?

Reply:
Yeah gonna raise/merge that PR first